src/core/ext/xds/xds_client_grpc.cc

```diff
@@ -267,11 +259,9 @@ GrpcXdsClient::GrpcXdsClient(
 
 void GrpcXdsClient::Orphan() {
   MutexLock lock(g_mu);
-  if (g_xds_client_map != nullptr && g_xds_client_map->size() > 1) {
+  auto it = g_xds_client_map->find(key_);
+  if (it != g_xds_client_map->end() && it->second == this) {
     g_xds_client_map->erase(g_xds_client_map->find(key_));
```

Comment:
This should be `g_xds_client_map->erase(it)`.  We've already done the lookup, so there's no reason to do it again.